### PR TITLE
CORE-7353 refactor of logging and filtering of expired session events

### PR DESCRIPTION
**Pull request by LWogan**
Filter session events that have expired their TTL. 
Include further details in the MDC info to help debug issues in the future such as flow id and session id. 
Adjust log levels of inconsequential log events 

https://github.com/corda/corda-api/pull/720

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -40,7 +40,7 @@ bouncycastleVersion=1.70
 # Corda API libs revision (change in 4th digit indicates a breaking change)
 # Change to 5.0.0.xx-SNAPSHOT to pick up maven local published copy
 #cordaApiVersion=5.0.0.xxx-SNAPSHOT
-cordaApiVersion=5.0.0.510-beta+
+cordaApiVersion=5.0.0.511-alpha-1670318487569
```
Comment (fowlerrr) [inline gradle.properties]: Uh...

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowMDCServiceImpl.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                val startContext = payload.startContext
+                val startKey = startContext.statusKey
+                val holdingIdentityShortHash = startKey.identity.toCorda().shortHash.toString()
+                mapOf(
+                    MDC_VNODE_ID to holdingIdentityShortHash,
+                    MDC_CLIENT_ID to startContext.requestId,
+                    MDC_FLOW_ID to startKey.id
+                )
+            }
+            is SessionEvent -> {
```
Comment (fowlerrr) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowMDCServiceImpl.kt:50]: Is this missing `SessionInit` from the removed section above?

Comment (LWogan) [reply components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowMDCServiceImpl.kt:50]: no ive removed that restriction. In theory it should be the only SessionInits that hit here but when another SessionEvent e arrives with a null state we'll want to know the sessionId that caused it.

File: components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt [hunk 3]
```diff
@@ -25,17 +30,56 @@ class FlowMapperMessageProcessor(
         private val logger = contextLogger()
     }
 
+    private val sessionP2PTtl = flowConfig.getLong(FlowConfig.SESSION_P2P_TTL)
```
Comment (JamesHR3) [inline components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt:35]: Presumably we recreate the full message processor in the event that the flow config changes?

File: components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt [hunk 4]
```diff
@@ -25,17 +30,56 @@ class FlowMapperMessageProcessor(
...
     ): StateAndEventProcessor.Response<FlowMapperState> {
         val key = event.key
         logger.trace { "Received event: key: $key event: ${event.value}" }
         val value = event.value ?: return StateAndEventProcessor.Response(state, emptyList())
-        val executor = flowMapperEventExecutorFactory.create(key, value, state, flowConfig)
-        val result = executor.execute()
 
-        return StateAndEventProcessor.Response(result.flowMapperState, result.outputEvents)
+        return if (isExpiredSessionEvent(value) || !isValidState(state, value)) {
+            StateAndEventProcessor.Response(state, emptyList())
```
Comment (JamesHR3) [inline components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt]: Might be worth a debug/trace level log here just to say we've discarded the event.

File: components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt [hunk 5]
```diff
@@ -25,17 +30,56 @@ class FlowMapperMessageProcessor(
...
+            val result = executor.execute()
+            StateAndEventProcessor.Response(result.flowMapperState, result.outputEvents)
+        }
+    }
+
+    /**
+     * Only allow flow events to be processed when:
+     * - the messages are for a new state.
+     * - if it is an existing state that is set to [FlowMapperStateType.OPEN],
+     * - it is not a [FlowEvent]
```
Comment (JamesHR3) [inline components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt]: Slightly unclear on the last bullet in this list. Were you trying to say that we only process records that are `FlowEvent`s?

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/handlers/requests/FlowFailedRequestHandler.kt [hunk 6]
```diff
@@ -38,8 +38,8 @@ class FlowFailedRequestHandler @Activate constructor(
             FLOW_FAILED,
             request.exception.message ?: request.exception.javaClass.name
         )
-
-        val expiryTime = Instant.now().plusMillis(1.minutes.toMillis()).toEpochMilli() // TODO Should be configurable?
+        val flowCleanupTime = context.config.getLong(PROCESSING_FLOW_CLEANUP_TIME)
```
Comment (JamesHR3) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/handlers/requests/FlowFailedRequestHandler.kt:41]: FYI @conalsmith-r3 - we're making this configurable in this PR, so you'll need to update your upgrade PR accordingly.

File: components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt [hunk 7]
```diff
@@ -25,17 +30,56 @@ class FlowMapperMessageProcessor(
...
+        val payload = event.payload
+        if (payload is SessionEvent) {
+            val sessionEventExpiryTime = payload.timestamp.toEpochMilli() + sessionP2PTtl
+            val currentTime = Instant.now().toEpochMilli()
+            if (currentTime > sessionEventExpiryTime) {
+                return true
+            }
+        }
+
+        return false
```
Comment (JamesHR3) [inline components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt]: Do we have a unit test for this class? I wouldn't mind putting some verification in that we properly handle invalid states and expired session events.

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/factory/impl/FlowEventProcessorFactoryImpl.kt [hunk 8]
```diff
@@ -21,15 +22,19 @@ class FlowEventProcessorFactoryImpl @Activate constructor(
     @Reference(service = FlowEventExceptionProcessor::class)
     private val flowEventExceptionProcessor: FlowEventExceptionProcessor,
     @Reference(service = FlowEventContextConverter::class)
-    private val flowEventContextConverter: FlowEventContextConverter
+    private val flowEventContextConverter: FlowEventContextConverter,
+    @Reference(service = FlowMDCService::class)
+    private val flowMDCService: FlowMDCService
+
```
Comment (JamesHR3) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/factory/impl/FlowEventProcessorFactoryImpl.kt]: Incredibly minor but might as well take this whitespace out.

File: components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt (hunk 5 above)
Comment (LWogan) [reply components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt]: oops that kdoc changed a few too many times. ill update it. The mapper needs to make sure it processes the cleanup events when the state is CLOSING. it should only filter FlowEvents when closing.

File: components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt (hunk 7 above)
Comment (LWogan) [reply components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt]: yeah ive some on the way

File: components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt (hunk 3 above)
Comment (LWogan) [reply components/flow/flow-mapper-service/src/main/kotlin/net/corda/session/mapper/service/executor/FlowMapperMessageProcessor.kt:35]: yes

**Review by JamesHR3 — DISMISSED** [review]

**Review by fowlerrr — APPROVED** [review]

**Review by JamesHR3 — APPROVED** [review]

**Pull request merged**